training/test/test_website_crawler.py

```diff
@@ -0,0 +1,51 @@
+import unittest
+from data_loader.test.test_training_storer_logic import TEST_SPREADSHEET_ID
+from training.website_iterator import WebsiteIterator
+from data_loader import training_storer
+from training.website_crawler import WebsiteCrawler
+from daos.caching_manager import CachingManager
+from flask import Flask
+
+DEFAULT_PARA_SIZE = 5
+
+
+class TestWebsiteCrawler(unittest.TestCase):
+    def setUp(self):
+        self.app = Flask(__name__)
+        self.app.config['URL_TIMEOUT'] = 10
+        self.app_context = self.app.app_context()
+        self.app_context.push()
+        self.temp_caching_manager = CachingManager()  # A cache for the test
+
+        self.test_link = "https://www.10dakot.co.il/recipe/%D7%9E%D7%AA%D7%9B%D7%95%D7%9F-%D7%9C%D7%A4%D7%A0%D7%A7%D7%99%D7%99%D7%A7/"
+
+    def test_crawler(self):
+        original_spreadsheet = training_storer.get_values(ignore_un_tagged=False, cells_range='A:A', target_spreadsheet=TEST_SPREADSHEET_ID)
+        added_cells_count = WebsiteCrawler.crawl(self.test_link, caching_manager=self.temp_caching_manager,
+                                              paragraph_size=DEFAULT_PARA_SIZE)
+        updated_spreadsheet = training_storer.get_values(ignore_un_tagged=False, cells_range='A:A', target_spreadsheet=TEST_SPREADSHEET_ID)
+
+        # Suppose it's enough to compare the first rows and the last rows
+        assert len(updated_spreadsheet) == len(original_spreadsheet) + added_cells_count, "Spreadsheet length mismatch"
+        assert original_spreadsheet[0] == updated_spreadsheet[0] and original_spreadsheet[
+            len(original_spreadsheet)-1] == updated_spreadsheet[
+                   len(original_spreadsheet) - 1], "First rows or last rows don't match"
+
+        counter = 0
+        iterator = WebsiteIterator.get_iterator_for_website(self.test_link, caching_manager=self.temp_caching_manager, paragraph_size=DEFAULT_PARA_SIZE)
```

Comment:
Why are we testing internal logic of the crawler here?  I don't see any apparent reason

Reply:
This module should have been used as an integration test, which combines the **_website_iterator_** and **_training_storer_** modules. Although each one has a unit test, here we want to test the whole script without using mocks, unlike in a unit test. Specifically, we need to verify that when provided a URL, the data is correctly stored in the spreadsheet.
To accomplish this, we have to use an iterator to know the expected data to be seen in the updated spreadsheet.

I haven't found a better solution, but I would like to here your opinion.